cardano-node/cardano-node.cabal

```diff
@@ -64,16 +64,20 @@ library
                      , network-mux
                      , optparse-applicative
                      , ouroboros-consensus
+                     , ouroboros-consensus-byron
+                     , ouroboros-consensus-shelley
                      , ouroboros-consensus-cardano
                      , ouroboros-network
+                     , ouroboros-network-framework
                      , process
+                     , shelley-spec-ledger
```

Comment:
It's unfortunate that we have to list these dependencies here. The goal was that `ouroboros-consensus-cardano` would be enough for `cardano-node`.

I suppose an alternative would be to let `ouroboros-consensus-cardano` expose more.

Reply:
One possible solution would be move my `HasTxMaxSize` typeclass into `ouroboros-consensus-cardano`.

Reply:
We could have a class of ledger state functions that overlap between Byron and Shelley and put that in `ouroboros-consensus-cardano`. This example is just getting one of the protocol params (tx size) that has an instance for both ledgers.

Reply:
Yes, I expect more such type class will be needed in the future. Maybe there are already more such type classes in disguise in `cardano-node`?

Oh, I just realised: `RunNode` is *the* type class for such cross-block details. I think we should make `getMaxTxSize` part of `RunNode`.

Reply:
I did suggest that, but to get things working more quickly we went with this solution,

Reply:
I agree that it should be part of `RunNode`. We can revisit this and do that at some point.

Reply:
https://github.com/input-output-hk/ouroboros-network/issues/2106